### PR TITLE
chore: [#188081219] add trim to nospace and write-once-read-only widgets

### PR DESCRIPTION
**Pull request by FarazA22**
### Ticket

https://www.pivotaltracker.com/story/show/188081219

### Approach

Add trim() to the nospace and the write once widgets. Note: the write once widget name was updated to state no spaces can be added.

## Code author checklist

- [x] I have rebased this branch from the latest main branch
- [x] I have performed a self-review of my code
- [x] I have created and/or updated relevant documentation (EX: [Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po)), if necessary
- [x] I have not used any relative imports
- [x] I have checked for and removed instances of unused config from CMS
- [x] I have pruned any instances of unused code
- [x] I have not added any markdown to labels, titles and button text in config
- [x] If I added/updated any values in `userData` (including `profileData`, `formationData` etc), then I added a new migration file
- [x] If I added any new collections to the CMS config, then I updated the search tool and `cmsCollections.ts` (see [CMS Additions in Engineering Reference/FAQ](https://docs.google.com/document/d/1X4iWSGmBZdHYZ0jGqkwTR3_yyyqI_TiiJptfc8pi9Po/edit#heading=h.fu5jdsrcqxbh))
- [x] I have updated relevant `.env` values in both `.env-template` and in Bitwarden

### CODE REVIEW

File: web/src/pages/mgmt/cms.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ const CMS = dynamic(
       // @ts-expect-error: No type definition available
       CMS.init({ CMS_CONFIG });
       // @ts-expect-error: No type definition available
-      CMS.registerWidget("write-once-read-only", WriteOnceReadOnlyControl);
+      CMS.registerWidget("write-once-read-only-no-space", WriteOnceReadOnlyNoSpaceControl);
```
Comment (jphechter) [inline web/src/pages/mgmt/cms.tsx:66]: It looks like there are a couple other places where we still use `write-once-read-only`. If we're completely removing this widget, then we should update those to also use the new prop.

I just did a quick search in `config.yml` to be sure and saw a couple.

File: web/src/lib/cms/fields/writeoncereadonlyfield.tsx [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import React, { Component, createRef, ReactElement } from "react";
```
Comment (jphechter) [inline web/src/lib/cms/fields/writeoncereadonlyfield.tsx:1]: Should we rename this file?

Comment (FarazA22) [reply web/src/lib/cms/fields/writeoncereadonlyfield.tsx:1]: changed

File: web/src/pages/mgmt/cms.tsx (hunk 1 above)
Comment (FarazA22) [reply web/src/pages/mgmt/cms.tsx:66]: There were two, i missed them cuz I did a find with quotes included, they all need to be swapped

**Review by jphechter — APPROVED** [review]

**Pull request merged**